core/domain/activity_jobs_one_off.py

```diff
@@ -628,3 +629,110 @@ def reduce(key, values):
+    snapshot models.
+    """
+
+    # This list consists of the snapshot models whose associated commit log
+    # models are missing. The commit logs were found missing in a validation
+    # job.
+    SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS = [
+        exp_models.ExplorationSnapshotMetadataModel,
+        exp_models.ExplorationRightsSnapshotMetadataModel,
+        question_models.QuestionSnapshotMetadataModel,
+        skill_models.SkillSnapshotMetadataModel
+    ]
+    MODEL_NAMES_TO_PROPERTIES = {
+        'ExplorationSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'exploration-%s-%s',
+            'id_field': 'exploration_id'
+        },
+        'QuestionSnapshotMetadataModel': {
+            'parent_model_class': question_models.QuestionModel,
+            'commit_log_model_class': (
+                question_models.QuestionCommitLogEntryModel),
+            'id_string_format': 'question-%s-%s',
+            'id_field': 'question_id'
+        },
+        'SkillSnapshotMetadataModel': {
+            'parent_model_class': skill_models.SkillModel,
+            'commit_log_model_class': skill_models.SkillCommitLogEntryModel,
+            'id_string_format': 'skill-%s-%s',
+            'id_field': 'skill_id'
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s',
+            'id_field': 'exploration_id'
+
+        }
+    }
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        job_class = AddMissingCommitLogsJob
+        return job_class.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS
+
+    @staticmethod
+    def map(snapshot_model):
+        job_class = AddMissingCommitLogsJob
+        class_name = snapshot_model.__class__.__name__
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_properties = job_class.MODEL_NAMES_TO_PROPERTIES[class_name]
+        commit_log_id = (
+            model_properties['id_string_format'] % (model_id, version))
+        commit_log_model = (
+            model_properties['commit_log_model_class'].get_by_id(
+                commit_log_id))
+        commit_logs_should_exist = True
+
+        parent_model = (
+            model_properties['parent_model_class'].get_by_id(model_id))
+        if class_name == 'ExplorationRightsSnapshotMetadataModel':
+            if snapshot_model.commit_type in ['create', 'delete']:
+                commit_logs_should_exist = False
+
+        if commit_log_model is None and commit_logs_should_exist:
+            if parent_model is not None:
+                # Public is the default value for post_commit_status for the
```

Comment:
Bit concerned about defaulting to public, because if a new model class is added later, it's easy to miss that we need to be careful about how we set this.

Suggest pulling out the "post_commit_status" computation into its own logical section, and doing an allowlist check instead of a default. I.e. "if model of this type, do X, elif model of this type, do Y, else assert Z". So that, if a new type is added, something will fail.

Reply:
I have created a allowlist check now.